### PR TITLE
WIP Spring cleaning

### PR DESCRIPTION
**Pull request by jarrodwb**
There is a lot in this pull request and I think it may be a challenge to review. It maybe easiest to review the commits one at a time. The hardest will be the _layouts commit.

Everything builds locally and no broken links are introduces.

The main motivation for doing all of this is to make the site easier to understand and modify. It is pretty much just a large refactor with a few design changes to simplify things.

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -4,5 +4,14 @@ scalaversion: "2.12.0"
...
+exclude: [
+  bundle-vendor,
+  blog/README.md,
+  training/README.md,
+  README.md,
+  news/README.md,
+  events/README.md,
+  Gemfile,
+  Gemfile.lock
+  ]
```
Comment (ashawley) [inline _config.yml]: Or even simpler:

```
exclude:
  - bundle-vendor/
  - README.md
  - Gemfile
  - Gemfile.lock
```

Comment (jarrodwb) [inline _config.yml]: `*.md` knocks out nearly the whole site. I get the your point though. Your version is cleaner.

Comment (ashawley) [inline _config.yml]: > `*.md` knocks out nearly the whole site.

Sorry, it was untested.

File: download/all.md [hunk 2]
```diff
@@ -1,8 +1,26 @@
...
+      {% assign possibleVersion = possibleVersionShort | append:'.' %}
+    {% else %}
+      {% assign possibleVersion = possibleVersionShort %}
+    {% endif %}
+    {% for page in site.categories.download %}
+      {% assign releaseVersion = page.release_version | truncate:4, '' %}
+      {% if releaseVersion == possibleVersion %}
+        <div>
+          <a href="{{ page.url }}">{{ page.title }}</a>
+        </div>
```
Comment (ashawley) [inline download/all.md]: This HTML needs to start in the first column, instead it's being shown as source code:

```
<div>
  <a href="/download/2.12.0.html">Scala 2.12.0</a>
</div>
  
  
  
<div>
  <a href="/download/2.12.0-RC2.html">Scala 2.12.0-RC2</a>
</div>
  
...
```

**Comment by SethTisue** [comment]
Are we sure that _tools/redirections.txt is unused? Maybe @fsalvi uses it?

**Comment by jarrodwb** [comment]
I am not sure if it is still in use. I was going on the last modification date and taking a guess. At least it does not seem to be needed by Jekyll.

**Comment by SethTisue** [comment]
I don't know the templating stuff well enough to give a good technical review on that. But, other than my question about redirections.txt (hopefully Fabien will respond), LGTM.

**Comment by SethTisue** [comment]
Oh, now when I building I see:

```
[2016-11-21 08:38:52] ERROR `/webscripts/ajax/getFromTwitter.php' not found.
```

I don't recall seeing that error before.

**Comment by jarrodwb** [comment]
We can always role back a few commits if need be if the templates break something.

I think the /webscripts/ajax/... and a few other things are sitting on the production server. It would be great to get everything in one place. The last few steps from building the site and getting to production are still a bit mysterious to me. I am interested in learning more though.

**Comment by fsalvi** [comment]
About the building error, I guess it's because of the last modification I made:
https://github.com/scala/scala-lang/commit/3657212411f780655345322af3d2b8eba45b7e13

I just fixed it.

**Comment by fsalvi** [comment]
I couldn't see any script using _tools/redirections.txt to automatically update .htaccess
I guess it was used at first to not forget some important redirections... But afaik, it's not used anymore.

**Comment by fsalvi** [comment]
getFromTwitter.php is in a private repository on github because it contains keys and secrets for the API calls

**Comment by jarrodwb** [comment]
I rebased onto master. I guess I still have some things to learn. A ton of commits were added to this PR. The only changes I made are the last two.

**Comment by jarrodwb** [comment]
I am not sure what to do. It looks like merging this in may put a lot of redundant commits into the commit log. Does anyone know how to deal this this situation? I will done some more research and try to fix it.

**Comment by jarrodwb** [comment]
Okay I think the easiest thing to do was to just make a new branch and get rid of the rebase and the weird merge.

See #550

**Comment by ashawley** [comment]
Looks like your new commits are at the top, so it should be easy.  Something like this could fix it (where remote `jarrodwb` for me is probably `origin` for you):
```
git fetch jarrodwb
# original commit was 34f2333:
git checkout -B spring-cleaning 34f2333 
# get the 3 commits from the diverged branch:
git rebase --onto spring-cleaning jarrodwb/spring-cleaning~3 jarrodwb/spring-cleaning
# Detaches HEAD:
git status
# Re-attach:
git checkout -B spring-cleaning HEAD
git checkout master
git pull upstream master
git checkout -
git rebase master
git diff jarrodwb/spring-cleaning
```

There's no output on the last line, so you can effectively push up changes and know you are only fixing the version history but nothing else.

**Comment by jarrodwb** [comment]
Thanks Aaron for the information. It looks like that may work. I already moved things over to #550.

**Pull request closed without merging**